gtsam/hybrid/HybridGaussianFactorGraph.cpp

```diff
@@ -299,13 +339,40 @@ discreteElimination(const HybridGaussianFactorGraph &factors,
 #if GTSAM_HYBRID_TIMING
   gttic_(EliminateDiscrete);
 #endif
-  // NOTE: This does sum-product. For max-product, use EliminateForMPE.
-  auto result = EliminateDiscrete(dfg, frontalKeys);
+  // Check if separator is empty
+  Ordering allKeys(dfg.keyVector());
+  Ordering separator;
+  std::set_difference(allKeys.begin(), allKeys.end(), frontalKeys.begin(),
+                      frontalKeys.end(),
+                      std::inserter(separator, separator.begin()));
+
+  // If the separator is empty, we have a clique of all the discrete variables
+  // so we can use the TableFactor for efficiency.
+  if (separator.size() == 0) {
+    // Get product factor
+    TableFactor product = TableProduct(dfg);
+
+#if GTSAM_HYBRID_TIMING
+    gttic_(EliminateDiscreteFormDiscreteConditional);
+#endif
+    auto conditional = std::make_shared<DiscreteConditional>(
+        frontalKeys.size(), product.toDecisionTreeFactor());
 #if GTSAM_HYBRID_TIMING
-  gttoc_(EliminateDiscrete);
+    gttoc_(EliminateDiscreteFormDiscreteConditional);
 #endif
 
-  return {std::make_shared<HybridConditional>(result.first), result.second};
+    TableFactor::shared_ptr sum = product.sum(frontalKeys);
```

Comment:
I still need to compute `sum` as the $\tau$ factor to return.

Reply:
Yikes, then we compute the sum twice. Maybe have an output argument in TableDistribution constructor?

Reply:
That will have to be in the follow up PR that introduces the TableDistribution. I can add a factory constructor that returns the sum.